src/H5Tcompound.c

```diff
@@ -473,7 +473,12 @@ H5T__insert(H5T_t *parent, const char *name, size_t offset, const H5T_t *member)
     parent->shared->u.compnd.memb[idx].name   = H5MM_xstrdup(name);
     parent->shared->u.compnd.memb[idx].offset = offset;
     parent->shared->u.compnd.memb[idx].size   = total_size;
-    parent->shared->u.compnd.memb[idx].type   = H5T_copy(member, H5T_COPY_ALL);
+    {
+        H5T_t* t = H5T_copy(member, H5T_COPY_ALL);
```

Comment:
This variable seems unnecessary here. We could probably replace this block with:

`if (NULL == (parent->shared->u.compnd.memb[idx].type = H5T_copy(member, H5T_COPY_ALL)))
    HGOTO_ERROR(H5E_DATATYPE, H5E_CANTCOPY, FAIL, "member type cannot be copied")`

Reply:
Yes, seems ok